dev/io.openliberty.microprofile.telemetry.internal.common/src/io/openliberty/microprofile/telemetry/internal/interfaces/OpenTelemetryAccessor.java

```diff
@@ -14,31 +14,58 @@
 
-import io.openliberty.microprofile.telemetry.internal.common.OpenTelemetryInfo;
-import io.openliberty.microprofile.telemetry.internal.common.OpenTelemetryInfoFactory;
+import io.openliberty.microprofile.telemetry.internal.common.info.ErrorOpenTelemetryInfo;
+import io.openliberty.microprofile.telemetry.internal.common.info.OpenTelemetryInfo;
 import io.opentelemetry.api.baggage.Baggage;
 import io.opentelemetry.api.trace.Span;
 import io.opentelemetry.api.trace.Tracer;
+import io.optenliberty.microprofile.telemetry.internal.common.helpers.OSGIHelpers;
 
 public class OpenTelemetryAccessor {
 
     private static final TraceComponent tc = Tr.register(OpenTelemetryAccessor.class);
 
     //See https://github.com/open-telemetry/opentelemetry-java-docs/blob/main/otlp/src/main/java/io/opentelemetry/example/otlp/ExampleConfiguration.java
+    /**
+     * Gets or creates the instance of OpenTelemetry associated with this application and returns it wrapped inside an OpenTelemetryInfo.
+     *
+     * @return An instance of OpenTelemetryInfo containing the instance of OpenTelemetry associated with this application. This instance will be a no-op OpenTelemetry if telemetry
+     *         is disabled or the application has shut down.
+     */
     public static OpenTelemetryInfo getOpenTelemetryInfo() {
-        return OpenTelemetryInfoFactory.getOpenTelemetryInfo(ComponentMetaDataAccessorImpl.getComponentMetaDataAccessor().getComponentMetaData().getJ2EEName());
+        try {
+            OpenTelemetryInfoFactory factory = OSGIHelpers.getService(OpenTelemetryInfoFactory.class, OpenTelemetryAccessor.class);
+            return factory.getOpenTelemetryInfo();
+        } catch (Exception e) {
+            return new ErrorOpenTelemetryInfo();
+        }
     }
 
+    /**
+     * Gets or creates a tracer instance from the TracerProvider for the OpenTelemetry instance associated with this application.
+     *
+     * @return An tracer instance from the instance of OpenTelemetry associated with this application. This instance will be a no-op if telemetry is disabled or the application has
+     *         shut down.
+     */
     public static Tracer getTracer() {
-        return OpenTelemetryInfoFactory.getTracer(ComponentMetaDataAccessorImpl.getComponentMetaDataAccessor().getComponentMetaData().getJ2EEName());
+        return getOpenTelemetryInfo().getTracer();
     }
 
+    /**
+     * Acquires the span for the current Context
+     *
+     * @return Returns the Span from the current Context, falling back to a default, no-op Span if there is no span in the current context.
+     */
     public static Span getSpan() {
         return Span.current();
     }
 
+    /**
+     * Acquires the Baggage for the current Context
+     *
+     * @return Returns Baggage from the current Context, falling back to empty Baggage if none is in the current Context.
+     */
     public static Baggage getBaggage() {
         return Baggage.current();
     }
```

Comment:
As a follow-up: remove these three methods.

- `Tracer` is now available from `OpenTelemetryInfo`
- Callers which need the current span or baggage should call the OTel API directly. Having our own method which just delegates to the OTel API is confusing because it looks like we're doing something special when we're not.